### ISSUE TITLE
Synaptics Trackpad buttons do not work.

### ISSUE THREAD

**Issue by Shockshwat** [acidanthera/bugtracker#1970]
└── Trackpad (PS2)
      ├── Device ID: 0xffff
      └── Vendor: 0x5ac

I am using SynapticsPS2 Trackpad with the above specifications. After using Rehabman I got to know about this tool and when I got it, I realized that the trackpad buttons do not work. Is there a fix for it or anything?

**Comment by Core-i99** [comment]
I've got the same issue. But using voodoosmbus kexts the buttons now work. (I need to look which kexts I used exactly)

**Comment by Shockshwat** [comment]
> 
But dude i have PS2 port not I2c

**Comment by Core-i99** [comment]
> > 
> But dude i have PS2 port not I2c
> 

But dude it's not that your trackpad is ps2 just like mine that you won't need smbus kexts

**Comment by Shockshwat** [comment]
My trackpad doesn't show up in the compatible list?

**Comment by Shockshwat** [comment]
As i thought 
It didn't work sadly

**Comment by kyeng-vis** [comment]
Hey @Shockshwat, can you try VoodooPS2Controller 2.1.6?

**Comment by Shockshwat** [comment]
@kyeng-vis It still doesnt work it seems

**Comment by kyeng-vis** [comment]
seem like we are having different issue, I am stuck with 2.1.6, anything newer then my trackpad stop working

**Comment by MiddlemanTM** [comment]
> └── Trackpad (PS2) ├── Device ID: 0xffff └── Vendor: 0x5ac
> 
> I am using SynapticsPS2 Trackpad with the above specifications on HP ProBook 4440s Ivy Bridge. After using RehabmanPs2 I got to know about VoodooPS2 by acidenthra tool and when I got it, I realized that the trackpad buttons do not work. Is there a fix for it or anything?

Yes there is. Fix has to be placed like this (in this exact order). Note the trackpad kext itself is disabled.:
![Screen Shot 2022-03-24 at 12 12 37 AM](https://user-images.githubusercontent.com/78093030/161978370-a2d32a8e-7d8e-442a-885e-84ca919dd43f.png)

You will also need SSDT-PLUG & SSDT-XOSI and these patches (_OSI to XOSI rename):

![Screen Shot 2022-04-06 at 8 56 18 PM](https://user-images.githubusercontent.com/78093030/161979683-c6c1614c-af5e-4036-a870-a39a29de576e.png)

You can find the files here (it's a build for Acer Nitro 5 but kext arrangement can also work with HP as I have another HP laptop that uses this exact setup) > https://www.tonymacx86.com/threads/guide-oc-monterey-acer-nitro-5-an515-52-core-i7-8750h-samsung-1tb-960-evo-pcie-nvme.319629/

**Comment by andreszerocross** [comment]
> > └── Trackpad (PS2) ├── Device ID: 0xffff └── Vendor: 0x5ac
> > I am using SynapticsPS2 Trackpad with the above specifications on HP ProBook 4440s Ivy Bridge. After using RehabmanPs2 I got to know about VoodooPS2 by acidenthra tool and when I got it, I realized that the trackpad buttons do not work. Is there a fix for it or anything?
> 
> Yes there is. Fix has to be placed like this (in this exact order). Note the trackpad kext itself is disabled.: ![Screen Shot 2022-03-24 at 12 12 37 AM](https://user-images.githubusercontent.com/78093030/161978370-a2d32a8e-7d8e-442a-885e-84ca919dd43f.png)
> 
> You will also need SSDT-PLUG & SSDT-XOSI and these patches (_OSI to XOSI rename):
> 
> ![Screen Shot 2022-04-06 at 8 56 18 PM](https://user-images.githubusercontent.com/78093030/161979683-c6c1614c-af5e-4036-a870-a39a29de576e.png)
> 
> You can find the files here (it's a build for Acer Nitro 5 but kext arrangement can also work with HP as I have another HP laptop that uses this exact setup) > https://www.tonymacx86.com/threads/guide-oc-monterey-acer-nitro-5-an515-52-core-i7-8750h-samsung-1tb-960-evo-pcie-nvme.319629/

Your picture show that you're using I2C trackpad not PS2 Trackpad. PS2 Trackpad didn't use VoodooI2C.

**Comment by MiddlemanTM** [comment]
> > > └── Trackpad (PS2) ├── Device ID: 0xffff └── Vendor: 0x5ac
> > > I am using SynapticsPS2 Trackpad with the above specifications on HP ProBook 4440s Ivy Bridge. After using RehabmanPs2 I got to know about VoodooPS2 by acidenthra tool and when I got it, I realized that the trackpad buttons do not work. Is there a fix for it or anything?
> > 
> > 
> > Yes there is. Fix has to be placed like this (in this exact order). Note the trackpad kext itself is disabled.: ![Screen Shot 2022-03-24 at 12 12 37 AM](https://user-images.githubusercontent.com/78093030/161978370-a2d32a8e-7d8e-442a-885e-84ca919dd43f.png)
> > You will also need SSDT-PLUG & SSDT-XOSI and these patches (_OSI to XOSI rename):
> > ![Screen Shot 2022-04-06 at 8 56 18 PM](https://user-images.githubusercontent.com/78093030/161979683-c6c1614c-af5e-4036-a870-a39a29de576e.png)
> > You can find the files here (it's a build for Acer Nitro 5 but kext arrangement can also work with HP as I have another HP laptop that uses this exact setup) > https://www.tonymacx86.com/threads/guide-oc-monterey-acer-nitro-5-an515-52-core-i7-8750h-samsung-1tb-960-evo-pcie-nvme.319629/
> 
> Your picture show that you're using I2C trackpad not PS2 Trackpad. PS2 Trackpad didn't use VoodooI2C.

Alright then you win lol. But at least the trackpads work. I guess this is a solution for those with I2C chipsets then. But for sure it also works with older Kaby Lake/Whiskey Lake HP trackpads.

**Comment by 1Revenger1** [comment]
I'm wondering if it's even I2C. I know quite a few of the ProBooks use SMBus trackpads instead. May be worth checking ioreg to see what is attached to what. If the trackpad is I2C, then having VoodooPS2Mouse attach to it could cause additional problems.

**Comment by andreszerocross** [comment]
> I'm wondering if it's even I2C. I know quite a few of the ProBooks use SMBus trackpads instead. May be worth checking ioreg to see what is attached to what. If the trackpad is I2C, then having VoodooPS2Mouse attach to it could cause additional problems.

Correct, i curious it's PS2 too then it mean VoodooI2C and plugin is not loaded.

However he is disable VoodooPS2Trackpad, some Synaptic can work with VoodooPS2Mouse plugin. If it's true then he (MiddlemanTm) can't see Trackpad prefeferences in system preference. Because default VoodooPS2Mouse will recognize Trackpad as mouse device so no trackpad preferences.

But, it will be good if he can attach IOREG so we can see the detail

**Comment by yusufklncc** [comment]
My Synaptic PS2 touchpad and clickpad work fine but mine prtsc key is not working.

**Comment by Shockshwat** [comment]
only clicks arent working otherwise the trackpad is fine

**Comment by yusufklncc** [comment]
yeah i know. But i am trying to work them. I am using SSDT-Clickpad and VoodooPS2Controller (Rehabman 2018 version). Trackpad and clicks works fine but my printscreen key is not working. When i use acidanthera voodoops2, trackpad and keys are working fine but all clicks are right clicking.

**Comment by MiddlemanTM** [comment]
Okay folks looks like I've finally sorted out how the trackpad should work (including gestures). If it helps try the settings like this. This is confirmed working on my Acer Nitro 5 AN515-52 build under Monterey 12.3. Also disable SSDT-GPIO and SSDT-I2C if you have them: 

![Screen Shot 2022-04-18 at 8 08 51 AM](https://user-images.githubusercontent.com/78093030/163737344-fbc9138c-6c9c-477f-8cc8-b1c07e767697.png)

You will also need the latest files from Acidanthera for VoodooI2C, VoodooI2CHID & VoodooPS2Controller.
As for I2C or PS2, my Windows shows the touchpad is a I2CHID device (on Acer). The keyboard is PS2, hence the dual requirement of both drivers.

About the IOReg file from the older HP setup I'll see what I can do. Will have to be tomorrow though.

**Comment by SomeName6** [comment]
> Okay folks looks like I've finally sorted out how the trackpad should work (including gestures). If it helps try the settings like this. This is confirmed working on my Acer Nitro 5 AN515-52 build under Monterey 12.3. Also disable SSDT-GPIO and SSDT-I2C if you have them:
> 
> ![Screen Shot 2022-04-18 at 8 08 51 AM](https://user-images.githubusercontent.com/78093030/163737344-fbc9138c-6c9c-477f-8cc8-b1c07e767697.png)
> 
> You will also need the latest files from Acidanthera for VoodooI2C, VoodooI2CHID & VoodooPS2Controller. As for I2C or PS2, my Windows shows the touchpad is a I2CHID device (on Acer). The keyboard is PS2, hence the dual requirement of both drivers.
> 
> About the IOReg file from the older HP setup I'll see what I can do. Will have to be tomorrow though.

No comment
